src/decode.ts

```diff
@@ -95,39 +129,31 @@ export function deserialize(qr: string): DataModel {
 			payment.bankAccounts.push(account);
 		}
 
-		data.shift(); // StandingOrderExt
-		data.shift(); // DirectDebitExt
-
-		// narrowing payment type
-		switch (payment.type) {
-			case PaymentOptions.PaymentOrder:
-				break;
-
-			case PaymentOptions.StandingOrder:
-				payment = {
-					...payment,
-					day: Number(data.shift()) as Day,
-					month: Number(data.shift()),
-					periodicity: data.shift() as Periodicity,
-					lastDate: data.shift(),
-				} satisfies StandingOrder;
-				break;
-
-			case PaymentOptions.DirectDebit:
-				payment = {
-					...payment,
-					directDebitScheme: Number(data.shift()),
-					directDebitType: Number(data.shift()),
-					mandateId: data.shift(),
-					creditorId: data.shift(),
-					contractId: data.shift(),
-					maxAmount: Number(data.shift()),
-					validTillDate: data.shift(),
-				} satisfies DirectDebit;
-				break;
-
-			default:
-				break;
+		const standingOrderExt = data.shift(); // StandingOrderExt
+		if (standingOrderExt === "1" && payment.type === PaymentOptions.StandingOrder) {
+			payment = {
+				...payment,
+				day: decodeNumber(data.shift()),
+				month: decodeNumber(data.shift()),
+				periodicity: decodeString(data.shift()) as Periodicity | undefined,
+				lastDate: decodeString(data.shift()),
+			} satisfies StandingOrder;
+		}
+		const directDebitExt = data.shift(); // DirectDebitExt
+		if (directDebitExt === "1" && payment.type === PaymentOptions.DirectDebit) {
+			payment = {
+				...payment,
+				directDebitScheme: decodeNumber(data.shift()),
+				directDebitType: decodeNumber(data.shift()),
+				variableSymbol: decodeString(data.shift()),
```

Comment:
I improved the decoding of DirectDebit. Still, I wasn't able to complete it, because I think the documentation is unclear:
<img width="488" alt="image" src="https://github.com/user-attachments/assets/854d23fa-3864-4676-8ab6-0ca68b4aecf3">
In my opinion, it's unclear, how to encode/decode those 3 variants.